---
title: "COV_assembly"
author: "Rashedul"
date: "7/14/2020"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Spike variants

```{r, cache=TRUE, warning=FALSE, message=FALSE}
library(tidyverse)
setwd("~/Documents/research/asm/covid19-Assembly/plots/")

s = read.table("~/Documents/research/asm/covid19-Assembly/files/SpikeSNVsDetails.txt")
s2 = data.frame(str_split_fixed(s$V1, "_", 3), s)

#number of samples used for variant calling
length(unique(s2$X1))

s3 = s2 %>% group_by(X2, V5) %>%
  summarise(n = n())

ggplot(s3, aes(X2, n, fill = V5)) +
  geom_bar(position="stack", stat="identity") + 
    xlab("") +
    ylab("Total variants in 347 samples at Spike locus") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "right")

ggsave("Variant_count_spike.pdf", width = 18, height = 13, units = "cm", device = 'pdf')

# mega and meta had 100% spike
t = read.table("~/Documents/research/asm/covid19-Assembly/files/intersected_samples_list.txt")

mgs = read.table("~/Documents/research/asm/covid19-Assembly/files/spike_intersected_megahit_SRR.txt")
table(duplicated(mgs))
mts = read.table("~/Documents/research/asm/covid19-Assembly/files/spike_intersected_metaspades_SRR.txt")
table(duplicated(mts))
mts2 = data.frame(V1 = mts[!duplicated(mts$V1), ])
length(unique(intersect(mgs$V1, mts2$V1)))

se_temp = s2 %>% select(X1, X2, V5)
mg = left_join(mgs, se_temp, by = c("V1" = "X1")) %>% filter(X2 %in% c("megahit"))
mt = left_join(mts2, se_temp, by = c("V1" = "X1")) %>% filter(X2 %in% c( "metaspades"))

length(unique(intersect(mg$V1, mt$V1)))

ms_common = inner_join(mg, mt, by = c("V1" = "V1")) 

s3 %>% filter(X2 %in% c("megahit", "metaspades")) %>%
ggplot(aes(X2, n, fill = V5)) +
  geom_bar(position="stack", stat="identity") + 
    xlab("") +
    ylab("Total variants at Spike locus") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "right")

ggsave("Variant_count_spike_mega_meta.pdf", width = 10, height = 13, units = "cm", device = 'pdf')



## total count of variants
v = read.table("~/Documents/research/asm/covid19-Assembly/files/Variant_count_all_assembler.txt")

ggplot(v, aes(V1, V2, fill = V1)) +
  geom_bar(position="stack", stat="identity") + 
    xlab("") +
    ylab("Total variants in 416 samples") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none")

ggsave("Variant_count_all.pdf", width = 18, height = 13, units = "cm", device = 'pdf')

# correlation variant_count and %GF
x = read_tsv("~/Documents/research/asm/covid19-Assembly/files/report_metaquast_PE_all.tsv")

x2 = melt(x, id.vars = "Assembly")
x2$Assembly = gsub("# ", "", x2$Assembly)
x3 = data.frame(str_split_fixed(x2$variable, "_", 3), value = as.numeric(x2$value), x2)

x4 = x3 %>% filter(Assembly == "Genome fraction (%)") %>% na.omit() %>%
  group_by(X2) %>%
  summarise(mean = mean(value))

cor(v$V2, x4$mean)
#0.734553

# variants per sample
df = read_csv("~/Documents/research/asm/covid19-Assembly/files/sample_variants_count.csv")
# replace -32 with NA
df[df == -32] <- NA
df2 = melt(df)

ggplot(df2, aes(variable, value)) +
  geom_boxplot() + 
    xlab("") +
    ylab("Variants per sample") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none")

ggsave("Variant_count_per_sample.pdf", width = 18, height = 13, units = "cm", device = 'pdf')


df3 = df2 %>% 
  group_by(variable) %>%
  summarise(mean = mean(value, na.rm = T),
            median = median(value, na.rm = T),
            sd = sd(value, na.rm = T))

ggplot(df3, aes(variable, mean)) + 
    geom_col() +  
    geom_errorbar(aes(ymin = mean - sd, ymax = mean + sd), width=0.2) +
    xlab("") +
    ylab("Variants per sample") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none")

print(df3, n = Inf)
```

### CPU and RAM

```{r, cache=TRUE, warning=FALSE, message=FALSE}
library(tidyverse)
setwd("~/Documents/research/asm/covid19-Assembly/plots/")

c = read_csv("~/Documents/research/asm/covid19-Assembly/files/CPU_time.csv")
ggplot(c, aes(Assembler, `Time(s)`)) +
    geom_jitter(alpha = .5, width = .1, aes(color = Assembler)) +
    geom_boxplot(alpha = 0.3) +
    #scale_color_viridis(discrete=TRUE) +
    #scale_fill_material("red") +
    xlab("") +
    #ylab(variableToPlot) +
    #facet_grid(Assay_Type~.) +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none")

ggsave("CPU_times.pdf", width = 18, height = 13, units = "cm", device = 'pdf')

#ram
c = read_csv("~/Documents/research/asm/covid19-Assembly/files/RAM.csv")
ggplot(c, aes(Assembler, Ram_percentage)) +
    geom_jitter(alpha = .5, width = .1, aes(color = Assembler)) +
    geom_boxplot(alpha = 0.3) +
    #scale_color_viridis(discrete=TRUE) +
    #scale_fill_material("red") +
    xlab("") +
    #ylab(variableToPlot) +
    #facet_grid(Assay_Type~.) +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none")

ggsave("RAM_usage.pdf", width = 18, height = 13, units = "cm", device = 'pdf')
```

### variant heatmap

```{r, cache=TRUE, warning=FALSE, message=FALSE}
library(pheatmap)
library(tidyverse)

meta = read_csv("~/Documents/research/asm/covid19-Assembly/files/PE_561samples_final.csv")
mh = read.table("~/Documents/research/asm/covid19-Assembly/files/megahit_alignment_matrix_50bp.tsv", head = F)
ms = read.table("~/Documents/research/asm/covid19-Assembly/files/metaspade_alignment_matrix_50bp.tsv", head = F)
ab = read.table("~/Documents/research/asm/covid19-Assembly/files/abyss63_alignment_matrix_50bp.tsv", head = F)
tr = read.table("~/Documents/research/asm/covid19-Assembly/files/trinity_alignment_matrix_50bp.tsv", head = F)

hs = cbind(mh, ms, ab, tr)
hst = data.frame(t(hs))
hst2 = hst %>% filter(X1 != "A.ID.megahit.50bp_overlap") %>% 
    filter(X1 != "A.ID.metaspade.50bp_overlap") %>% 
    filter(X1 != "A.ID.abyss63.50bp_overlap") %>% 
    filter(X1 != "A.ID.trinity.50bp_overlap")

hst3 = data.frame(str_split_fixed(hst2$X1, "_", 3), hst2)

megahit = hst3 %>% filter(hst3$X2.1 == "megahit") %>% select(-X2.1) %>% select(-X3.1) %>% select(-X1)
metaspades = hst3 %>% filter(hst3$X2.1 == "metaspades") %>% select(-X2.1) %>% select(-X3.1) %>% select(-X1)
abyss = hst3 %>% filter(hst3$X2.1 == "abyss63") %>% select(-X2.1) %>% select(-X3.1) %>% select(-X1)
trinity = hst3 %>% filter(hst3$X2.1 == "trinity") %>% select(-X2.1) %>% select(-X3.1) %>% select(-X1)

both = merge(megahit, metaspades, by.x = "X1.1", by.y = "X1.1") 
both = inner_join(megahit, metaspades, by = c("X1.1" = "X1.1")) %>% 
    inner_join(abyss, by = c("X1.1" = "X1.1")) %>%
    inner_join(trinity, by = c("X1.1" = "X1.1"))

both2 = inner_join(meta, both, by = c( "Run" = "X1.1"))
both3 = both2[, c(2,6, 15:ncol(both2))] 
both3 %>% group_by(Assay_Type) %>%
    summarise(n = n())

#write.table(both3, "~/Documents/research/asm/covid19-Assembly/files/megahit_metaspades_heatmap_matrix.tsv", sep = "\t", quote = F, row.names = F, col.names = F)

both4 = data.frame(both3[,3:ncol(both3)]) 

#conver factor to numeric
both5 = both4 %>% 
  mutate_all(~as.numeric(as.character(.)))
rownames(both5) = both3$Run

#pheatmap(both5, cluster_cols = F, cluster_rows = F, show_rownames = F, show_colnames = F, color = colorRampPalette(c("red", "white", "gray"))(1000), border_color = NA, gaps_col = 599)

#add annotation
anno = data.frame(Assay = both3$Assay_Type)
rownames(anno) = rownames(both5)

pheatmap(both5, annotation_row = anno, cluster_cols = F, cluster_rows = F, show_rownames = F, show_colnames = F, color = colorRampPalette(c("red", "white", "gray"))(1000), border_color = NA, gaps_col = c(599, 599+599, 599+599+599), gaps_row = c(82, 82+65, 82+65+58, 82+65+58+88))

plot.new()
jpeg("~/Documents/research/asm/covid19-Assembly/plots/heatmap_v2.jpg", width = 1800, height = 1000)
pheatmap(both5, annotation_row = anno, cluster_cols = F, cluster_rows = F, show_rownames = F, show_colnames = F, color = colorRampPalette(c("red", "white", "gray"))(1000), border_color = NA, gaps_col = c(599, 599+599, 599+599+599), gaps_row = c(81, 81+60, 81+60+41, 81+60+41+83))
dev.off()

#get best and worst asm
worst = data.frame(asm = rowSums(both5)) %>% arrange(asm) %>% head() 
best = data.frame(asm = rowSums(both5)) %>% arrange(asm) %>% tail()
write.table(rbind(worst, best), "~/Documents/research/asm/covid19-Assembly/files/wrostAndBest_asm.tsv", col.names = F, quote = F)

#heatmap of worst and best
make_heatmap = function(SRR) {
  SRR12182155 = hst3 %>% filter(hst3$X1.1 == SRR)  %>% select(-X3.1) %>% select(-X1)
  SRR12182155_v2 = SRR12182155 %>% select(-X1.1) %>% select(-X2.1)
  #conver factor to numeric
  SRR12182155_v3 = SRR12182155_v2 %>% 
  mutate_all(~as.numeric(as.character(.)))
  rownames(SRR12182155_v3) = SRR12182155$X2.1

#
pheatmap(SRR12182155_v3, cluster_cols = F, cluster_rows = F, show_rownames = T, show_colnames = F, color = colorRampPalette(c("red", "white", "gray"))(1000), border_color = "black", gaps_row = c(1:3))
}

#worst 3
plot.new()
pdf("SRR11783612.pdf", width = 18, height = 2)
make_heatmap("SRR11783612") #Targeted-Capture
dev.off()

plot.new()
pdf("SRR11954291.pdf", width = 18, height = 2)
make_heatmap("SRR11954291") #RNA-seq; 4th
dev.off()

make_heatmap("SRR11783580") #Targeted-Capture
make_heatmap("ERR4321689") #WGA


#best 3
plot.new()
pdf("SRR12182155.pdf", width = 18, height = 2)
make_heatmap("SRR12182155") #OTHER
dev.off()

plot.new()
pdf("SRR11903415.pdf", width = 18, height = 2)
make_heatmap("SRR11903415") #Targeted-Capture; 6th and rest of them are Other
dev.off()

make_heatmap("SRR12182119") #OTHER
make_heatmap("SRR12182157") #OTHER
make_heatmap("SRR11903415") #Targeted-Capture; 6th and rest of them are Other


SRR12182155 = hst3 %>% filter(hst3$X1.1 == "SRR12182155")  %>% select(-X3.1) %>% select(-X1)
SRR12182155_v2 = SRR12182155 %>% select(-X1.1) %>% select(-X2.1)
#conver factor to numeric
SRR12182155_v3 = SRR12182155_v2 %>% 
  mutate_all(~as.numeric(as.character(.)))
rownames(SRR12182155_v3) = SRR12182155$X2.1

pheatmap(SRR12182155_v3, cluster_cols = F, cluster_rows = F, show_rownames = T, show_colnames = F, color = colorRampPalette(c("red", "white", "gray"))(1000), border_color = "black", gaps_row = c(1:3))


#make genome. hand draw
g = read.table("~/Documents/research/asm/covid19-Assembly/files/GeneBank_Genes.bed")
g2 = data.frame(gene = g$V4, len = g$V3-g$V2, sp = g$V1)

ggplot(g2, aes(sp, len, fill = gene)) + 
  geom_bar(stat = "identity", aes(alpha = .0), color = "black") +
  coord_flip()

```

### assembly quality

```{r , cache=TRUE, warning=FALSE, message=FALSE}
library(tidyverse)
library(reshape2)
library(viridis)
library(ggsci)

#x = read_tsv("~/Documents/research/asm/covid19-Assembly/files/backup/bioRxiv_1074_assembly_report_PE_amplicon_viralRNA.tsv")
x = read_tsv("~/Documents/research/asm/covid19-Assembly/files/report_metaquast_PE_all.tsv")

x2 = melt(x, id.vars = "Assembly")
x2$Assembly = gsub("# ", "", x2$Assembly)
x3 = data.frame(str_split_fixed(x2$variable, "_", 3), value = as.numeric(x2$value), x2)

# temp: query samples by %GF
temp = x3 %>% filter(Assembly == "Genome fraction (%)")  %>% arrange(-value)

#mega and mega
temp %>% filter(X2 == "megahit") %>% head(n=10)
temp %>% filter(X2 == "metaspades") %>% head(n=10)

temp %>% filter(value >= 70) %>% 
  group_by(X1) %>%
  summarise(n = n()) %>%
  arrange(-n)

# end temp

meta = read_csv("~/Documents/research/asm/covid19-Assembly/files/PE_561samples_final.csv")
meta2 = meta %>% select(Run, Assay_Type)

xm = left_join(x3, meta2, by = c("X1" = "Run"))

#number of samples in each category
xm %>% distinct(variable, .keep_all = T) %>% 
    filter(Assembly == "contigs (>= 0 bp)") %>%
    group_by(Assay_Type, X2) %>% 
    summarise(n = n()) %>% 
    filter(n == max(n)) %>%
    distinct(Assay_Type, .keep_all = T)

#
setwd("~/Documents/research/asm/covid19-Assembly/plots/")
make_boxplot = function(variableToPlot)
{
  x4 = xm %>% filter(Assembly == variableToPlot) 
  stat = x4 %>%
    group_by(Assay_Type, X2) %>%
    summarize(mean = mean(value, na.rm = TRUE),
            median = median(value, na.rm = TRUE))
  #kable(stat)
  
  print(stat, n = Inf)
  plot = ggplot(x4, aes(X2, value)) +
    geom_jitter(alpha = .5, width = .1, aes(color = X2)) +
    geom_boxplot(alpha = 0.1) +
    #scale_color_viridis(discrete=TRUE) +
    #scale_fill_material("red") +
    xlab("") +
    ylab(variableToPlot) +
    facet_grid(Assay_Type~.) +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none") 
  print(plot)
  
  filename = str_replace_all(variableToPlot, "[[:punct:]]", "")
  ggsave(filename=paste(filename,".pdf", sep="_"), width = 12, height = 18, units = "cm", device = 'pdf')
}

make_boxplot("Genome fraction (%)")
make_boxplot("Largest contig")
make_boxplot("Total length")
make_boxplot("contigs")
make_boxplot("contigs (>= 1000 bp)")
make_boxplot("Largest alignment")
make_boxplot("mismatches per 100 kbp")
make_boxplot("indels per 100 kbp")
make_boxplot("Total aligned length")
#make_boxplot("unaligned contigs")
make_boxplot("N50")
make_boxplot("N75")
#make_boxplot("L50")
make_boxplot("L75")
make_boxplot("NA50")
make_boxplot("NA75")
#make_boxplot("LA50")
make_boxplot("LA75")
make_boxplot("unaligned mis. contigs")
make_boxplot("N's per 100 kbp")


#n50
make_boxplot_v2 = function(variableToPlot)
{
  x4 = xm %>% filter(Assembly == variableToPlot) 
  stat = x4 %>%
    group_by(X2, Assay_Type) %>%
    summarize(mean = mean(value, na.rm = TRUE),
            median = median(value, na.rm = TRUE),
            maximum = max(value, na.rm = TRUE))
            
print(stat, n = Inf)
    
stat2 = x4 %>%
    group_by(X2, Assay_Type) %>%
    tally(value)

  print("print sum of all i.e., total")
  print(stat2, n = Inf)

  plot = ggplot(x4, aes(X2, value)) +
    geom_jitter(alpha = .5, width = .1, aes(color = X2)) +
    geom_boxplot(alpha = 0.1) +
    #scale_color_viridis(discrete=TRUE) +
    #scale_fill_material("red") +
    xlab("") +
    ylab(variableToPlot) +
    facet_grid(Assay_Type~., scales = "free") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "none") 
  print(plot)
  
  filename = str_replace_all(variableToPlot, "[[:punct:]]", "")
  ggsave(filename=paste(filename,".pdf", sep="_"), width = 12, height = 18, units = "cm", device = 'pdf')
}

#la50
make_boxplot_v2("L50")
make_boxplot_v2("LA50")
make_boxplot_v2("Duplication ratio")
make_boxplot_v2("misassemblies")

#num of mis asm
xm %>% filter(Assembly == "misassemblies") %>% na.omit() %>%  dim()
xm %>% filter(Assembly == "misassemblies") %>% na.omit() %>% filter(value == 0) %>% dim()

#num of dup ratio
xm %>% filter(Assembly == "Duplication ratio") %>% na.omit() %>% dim()
xm %>% filter(Assembly == "Duplication ratio") %>% na.omit() %>% filter(value == 1) %>% dim()

# scatter plot
dr = xm %>% filter(Assembly == "Duplication ratio") 
mis = xm %>% filter(Assembly == "misassemblies") 
s3 = inner_join(mis, dr, by = c("variable" = "variable")) %>% 
  select(X1.x, X2.x, value.x, value.y, Assay_Type.y) %>% na.omit()
colnames(s3) = c("SRR", "asm", "mis", "dupliR", "assay")


ggplot(s3, aes(mis, dupliR)) +
  geom_jitter(aes(color = asm), shape = 18, size = 2.5) +
    xlab("Misassemblies") +
    ylab("Duplication ratio") +
    facet_grid(~assay, scales = "free") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", hjust = 1),
        strip.background =element_rect(fill="white"),
        legend.position = "bottom") +
  guides(color = guide_legend(override.aes = list(size = 3)))

ggsave("Scatter_misasm_dupliR.pdf", width = 24, height = 10, units = "cm")

s3 %>% group_by(asm, assay) %>%
  summarise(count = n())

# genomic feature
x4 = xm %>% filter(Assembly == "genomic features") 
x4$value.1 = gsub("part", "", x4$value.1)
x5 = data.frame(str_split_fixed(x4$value.1,"\\+", 2), x4) 
x6 = data.frame(assembly = x5$X2, match = as.numeric(as.character(x5$X1.1)), mismatch = as.numeric(as.character(x5$X2.1))) 

# need to add assay_type
  ggplot(x6, aes(assembly, 100*(match/49))) +
    geom_boxplot(aes(color = assembly)) +
    geom_jitter(alpha = .5, width = .1, aes(color = assembly)) +
    xlab("") +
    ylab("% features mapped") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1)) 
  
ggsave("percent features mapped.pdf", width = 15, height = 10, units = "cm")


# average qualily across all samples
avg = xm %>% filter(Assembly %in% c("Genome fraction (%)", "Largest alignment", "Duplication ratio", "misassemblies", "N50", "NA50",  "L50", "LA50")) %>%
  group_by(Assembly, X2) %>%
  summarise(mean = mean(value, na.rm = T),
            median = median(value, na.rm = T),
            max = max(value, na.rm = T),
            min = min(value, na.rm = T),
            sd = sd(value, na.rm = T),
            count = n())

print(avg, n = Inf)

# supplementary file 1
med = xm %>% 
  group_by(Assembly, X2) %>%
  summarise(median = median(value, na.rm = T)) %>%
  spread(key = "X2", value = median, fill = NA)

print(med, n = Inf)

write.table(med, "~/Documents/research/asm/covid19-Assembly/Supplementary_files/Median_Assembly_Quality_assemblers.tsv", quote = F, row.names = F, sep = ",")

# k-mer quality across all samples
kmer = c("abyss21", "abyss63", "abyss99", "metavelvet21", "metavelvet63", "metavelvet99", "ray21", "ray63", "ray99", "velvet21", "velvet63", "velvet99")
xmk = xm %>% filter(X2 %in% kmer) 

#global averge
kmer_stats = xmk %>% 
  group_by(Assembly, X2) %>%
  summarise(mean = mean(value, na.rm = T),
            median = median(value, na.rm = T),
            max = max(value, na.rm = T),
            min = min(value, na.rm = T),
            sd = sd(value, na.rm = T),
            count = n())

print(kmer_stats, n = Inf)

kmer_stats %>% filter(Assembly %in% c("Genome fraction (%)", "Largest alignment", "Duplication ratio", "misassemblies", "N50", "NA50",  "L50", "LA50")) %>%
  ggplot( aes(X2, mean)) +
  geom_col() +
  facet_grid(Assembly~., scales = "free") +
  xlab("") +
  ylab("Mean") +
  theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 90, hjust = 1)) 
  
ggsave("k-mer_stats.pdf", width = 25, height = 40, units = "cm")

```

### 90% of the genome is covered by single contig

```{r, cache=TRUE, warning=FALSE, message=FALSE}
library(tidyverse)
setwd("~/Documents/research/asm/covid19-Assembly/plots/")

#90%=29903*.9
contig = xm %>% filter(Assembly == "Largest contig") %>% filter(value >=26912.7)
#
contig2 = contig %>% group_by(X2, Assay_Type) %>% summarise(count = n())
print(contig2, n = Inf)


ggplot(contig, aes(X2)) +
    geom_histogram(stat = "count") +
    xlab("") +
    ylab("Count") +
    facet_grid(~Assay_Type) +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        strip.background =element_rect(fill="white"),
        axis.text = element_text(color = "black", angle = 90, hjust = 1)) 
ggsave("90_percent_genome_single_contig.pdf", width = 20, height = 6, units = "cm")
```

### correlation read vs genome

```{r , cache=TRUE, warning=FALSE, message=FALSE}
library(tidyverse)
setwd("~/Documents/research/asm/covid19-Assembly/plots/")
#r = read.table("~/Documents/research/asm/covid19-Assembly/files//read_QC_matrix.txt")[,c(1,4)]
r = read.table("~/Documents/research/asm/covid19-Assembly/files/read_QC_matrix.txt")[,c(1,4)]

r2 = data.frame(id = str_split_fixed(r$V1, "_", 2), read = r$V4)

#sum every two rows of PE data
r3 = data.frame(id = unique(r2$id.1), read = (rowsum(r2[,3], as.integer(gl(nrow(r2), 2, nrow(r2))))))
rx = xm %>% filter(Assembly == "Genome fraction (%)")
rx2 = inner_join(r3, rx, by = c("id" = "X1"))
#rxm = left_join(rx2, meta2, by = c("id" = "Run"))
subset(is.na(rx2)) %>% dim()
subset(!is.na(rx2)) %>% dim()
# metavelve and ray have lots of NA genome frac values

ggplot(rx2, aes(value, read/1e6)) +
    geom_point(aes(color = X2), shape = 18) +
    geom_smooth(method='lm', formula= y~x) +
    xlab("Genome fraction (%)") +
    ylab("Number of reads (million)") + 
    facet_grid(~Assay_Type, scales = "free") +
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        strip.background =element_rect(fill="white"),
        axis.text = element_text(color = "black", angle = 0, hjust = 1),
        legend.position = "none") 
ggsave("readVsGenome.pdf", width = 20, height = 6, units = "cm")

#global cor
rx3 = na.omit(rx2)
cor(rx3$value, rx3$read, method = "spearman")

correlation = rx2 %>% filter(Assay_Type == "AMPLICON") %>% na.omit()
cor(correlation$value, correlation$read)
cor(correlation$value, correlation$read, method = "spearman")

correlation = rx2 %>% filter(Assay_Type == "Targeted-Capture") %>% na.omit()
cor(correlation$value, correlation$read)
cor(correlation$value, correlation$read, method = "spearman")

correlation = rx2 %>% filter(Assay_Type == "WGA") %>% na.omit()
cor(correlation$value, correlation$read)
cor(correlation$value, correlation$read, method = "spearman")

correlation = rx2 %>% filter(Assay_Type == "RNA-Seq") %>% na.omit()
cor(correlation$value, correlation$read)
cor(correlation$value, correlation$read, method = "spearman")

correlation = rx2 %>% filter(Assay_Type == "OTHER") %>% na.omit()
cor(correlation$value, correlation$read)
cor(correlation$value, correlation$read, method = "spearman")


#read dist
dist = rx2 %>% distinct(id, .keep_all = T)
ggplot(dist, aes(Assay_Type, log10(read))) +
    geom_boxplot(notch = T, width = .5) +
    geom_jitter(alpha = .5, width = .1) +
    xlab("") +
    ylab("log10(Number of reads)") + 
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        strip.background =element_rect(fill="white"),
        axis.text = element_text(color = "black", angle = 90, hjust = 1)) 
ggsave("readDist.pdf", width = 7, height = 10, units = "cm")


```


### sample to assembler

```{r, message = FALSE}
library(tidyverse)
setwd("~/Documents/research/asm/covid19-Assembly/plots/")

rx2 %>% filter(Assembly == "Genome fraction (%)") %>% na.omit() %>%
    ggplot(aes(fct_reorder(id, read), X2, color = X2)) +
    geom_point(aes(size = value, alpha = .5)) +
    ylab("") +
    xlab("Samples are sorted by number of reads") + 
    theme(panel.background = element_rect(fill = "white"),
        panel.border = element_rect(fill = NA, colour = "black", size = .5),
        axis.text = element_text(color = "black", angle = 0, hjust = 1),
        axis.text.x=element_blank()) 

ggsave("sampleVsassembler_color_bk.pdf", width = 30, height = 20, units = "cm")

# select 4 samples with high and low mean frac to plot in mvista
#high = SRR11578289 (97.4), SRR11597206 (94.4)
#low = SRR11828432, SRR11828424
#ref: MN908947.3

mvista = x3 %>% filter(Assembly == "Genome fraction (%)") %>% na.omit() %>%
    group_by(X1) %>%
    summarise(mean = mean(value)) %>% arrange(mean)

```

### dataset preparation

```{r, cache=T, message=F, warning=FALSE}
library(tidyverse)
library(knitr)

c = read_tsv("~/Documents/research/asm/covid19-Assembly/files/SraRunTable_COVID19_14.06.20.txt")

c2 = tibble(c$Platform, c$Run, c$SRA_Sample, c$Instrument, c$LibraryLayout, c$Assay_Type, c$LibrarySelection, c$LibrarySource, c$Organism, c$geo_loc_name, c$host, c$host_disease, c$Consent)

names(c2) <- gsub("c\\$", "", names(c2))
colSums(!is.na(c2))

#
#write_csv(c2, "~/Documents/research/asm/covid19-Assembly/files//COVID19_14.06.20_metadata_final.csv")

# summarise metadata
#colnames(c2)

#good code example: https://uc-r.github.io/descriptives_categorical
table3 <- table(c2$Instrument, c2$Assay_Type)
table3 <- table( c2$Assay_Type, c2$LibrarySource)
table3 <- table(c2$Assay_Type, c2$LibrarySource, c2$LibraryLayout)
ftable(table3)

# will add table paper
df = c2 %>% 
    group_by(Assay_Type, LibrarySource, LibraryLayout) %>%
    tally() 

kable(df, caption = "Summary of all data. This table will add table paper")
#write_tsv(df, "~/Documents/research/asm/covid19-Assembly/files//summary_data.tsv")

df2 = df %>% filter(LibrarySource == "VIRAL RNA")
kable(df2, caption = "Summary of VIRAL RNA data")
##subsample main paper
#PE
set.seed(2020)
a1 = c2 %>% filter(LibraryLayout == "PAIRED" & Assay_Type == "AMPLICON" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "PE: AMPLICON of VIRAL RNA") %>% sample_n(100)

set.seed(2020)
a2 = c2 %>% filter(LibraryLayout == "PAIRED" & Assay_Type == "OTHER" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "PE: OTHER of VIRAL RNA")

set.seed(2020)
a3 = c2 %>% filter(LibraryLayout == "PAIRED" & Assay_Type == "RNA-Seq" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "PE: RNA-Seq of VIRAL RNA") %>% sample_n(100)

set.seed(2020)
a4 = c2 %>% filter(LibraryLayout == "PAIRED" & Assay_Type == "Targeted-Capture" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "PE: Targeted-Capture of VIRAL RNA") %>% sample_n(100)

set.seed(2020)
a5 = c2 %>% filter(LibraryLayout == "PAIRED" & Assay_Type == "WGA" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "PE: WGA of VIRAL RNA") %>% sample_n(100)

set.seed(2020)
a6 = c2 %>% filter(LibraryLayout == "PAIRED" & Assay_Type == "WGS" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "PE: WGS of VIRAL RNA")

#SE
set.seed(2020)
b1 = c2 %>% filter(LibraryLayout == "SINGLE" & Assay_Type == "AMPLICON" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "SE: AMPLICON of VIRAL RNA") %>% sample_n(100)

set.seed(2020)
b2 = c2 %>% filter(LibraryLayout == "SINGLE" & Assay_Type == "RNA-Seq" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "SE: RNA-Seq of VIRAL RNA") %>% sample_n(100)

set.seed(2020)
b3 = c2 %>% filter(LibraryLayout == "SINGLE" & Assay_Type == "Targeted-Capture" & LibrarySource == "VIRAL RNA") %>%
    mutate(LibType = "SE: Targeted-Capture of VIRAL RNA") %>% sample_n(100)

#dataset
ab_pe = rbind(a1, a2, a3, a4, a5, a6)
ab_se = rbind(b1, b2, b3)
#write_csv(ab_pe, "~/Documents/research/asm/covid19-Assembly/files//PE_561samples_final.csv")
#write_csv(ab_se, "~/Documents/research/asm/covid19-Assembly/files//SE_300samples_final.csv")
#write.table(ab_pe$Run, "~/Documents/research/asm/covid19-Assembly/files//PE_561samples_final_561runs.txt", col.names = F, row.names = F, quote = F)
#write.table(ab_se$Run, "~/Documents/research/asm/covid19-Assembly/files//SE_300samples_final_300runs.txt", col.names = F, row.names = F, quote = F)
#write_csv(a1, "~/Documents/research/asm/covid19-Assembly/files//PE_100samples_amplicon_bioRxiv.csv")
#write.table(a1[,2], "~/Documents/research/asm/covid19-Assembly/files//PE_100samples_amplicon_bioRxiv_100runs.txt", col.names = F, row.names = F, quote = F)

# test ram and cpu
set.seed(2020)
#20 samples will be selected from amplicon with similar depth
#ram = a1 %>% sample_n(20)

ab = rbind(ab_pe, ab_se)
ab2 = ab %>% group_by(LibType) %>% tally()
kable(ab2, caption = "List of total 9 different categories. Maximum 100 samples are randomly selected")
    
```

